### PR TITLE
feat: bump vislib version and add available IDE

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

This pull request includes various changes to different files within the codebase. The most important changes include adding a new data class to the `RunVisualization` class, modifying the `_visualize` method to handle visualization configuration, and updating import statements to include the `LANGUAGE_KEY` constant.

Main interface changes:

* <a href="diffhunk://#diff-1123ed6d2f2d62a8f6387b9aa7be118bcef1e86d910aaad829493a9c3d7a8710R32-R42">`src/promptflow/promptflow/contracts/_run_management.py`</a>: Added a new data class `VisualizationConfig` with a single attribute `availableIDEList` to the `RunVisualization` class.
* <a href="diffhunk://#diff-a28fe272514a81f000f5c01ae75a1af29af457913a822204fe12444eddfce795L273-R278">`src/promptflow/promptflow/_sdk/operations/_run_operations.py`</a>: Modified the `_visualize` method to handle visualization configuration based on the language of the flow. <a href="diffhunk://#diff-a28fe272514a81f000f5c01ae75a1af29af457913a822204fe12444eddfce795L273-R278">[1]</a> <a href="diffhunk://#diff-a28fe272514a81f000f5c01ae75a1af29af457913a822204fe12444eddfce795R12-R14">[2]</a> <a href="diffhunk://#diff-a28fe272514a81f000f5c01ae75a1af29af457913a822204fe12444eddfce795L29-R32">[3]</a> <a href="diffhunk://#diff-a28fe272514a81f000f5c01ae75a1af29af457913a822204fe12444eddfce795L296-R314">[4]</a>

Constant and import updates:

* <a href="diffhunk://#diff-b353941bce91518c7f494112d2dd5088b73681eb66e6ff19294edb9c3ec05d0fL16-R16">`src/promptflow/promptflow/contracts/flow.py`</a>: Updated the import statement for the `FlowLanguage` constant and changed the key used to access the language value in the `data` dictionary. <a href="diffhunk://#diff-b353941bce91518c7f494112d2dd5088b73681eb66e6ff19294edb9c3ec05d0fL16-R16">[1]</a> <a href="diffhunk://#diff-b353941bce91518c7f494112d2dd5088b73681eb66e6ff19294edb9c3ec05d0fL593-R593">[2]</a>
* <a href="diffhunk://#diff-e3499cf713c66612d91723759de1c5076c22689b2ce2c89678f82de624b57c2cL67-R67">`src/promptflow/promptflow/_sdk/_submitter/run_submitter.py`</a>: Updated the `_run_bulk` method to check the language of the flow using the `LANGUAGE_KEY` constant and modified the import statement to include the `LANGUAGE_KEY` constant. <a href="diffhunk://#diff-e3499cf713c66612d91723759de1c5076c22689b2ce2c89678f82de624b57c2cL67-R67">[1]</a> <a href="diffhunk://#diff-e3499cf713c66612d91723759de1c5076c22689b2ce2c89678f82de624b57c2cL9-R9">[2]</a>
* <a href="diffhunk://#diff-9ffef684cefef289f51e58b0aff1cd6cfa42363ace309d07f36dfe6559c5b84dL69-R69">`src/promptflow/promptflow/_constants.py`</a>: Updated the version number of the `VIS_LIB_VERSION` constant and added a new constant `LANGUAGE_KEY` and the `AvailableIDE` class. <a href="diffhunk://#diff-9ffef684cefef289f51e58b0aff1cd6cfa42363ace309d07f36dfe6559c5b84dL69-R69">[1]</a> <a href="diffhunk://#diff-0d3cf5f31883ff073bf1e11cb2e17db5cc8fe6e5e38cbb4af7b99f37ac31d41aR18">[2]</a> <a href="diffhunk://#diff-0d3cf5f31883ff073bf1e11cb2e17db5cc8fe6e5e38cbb4af7b99f37ac31d41aR34-R38">[3]</a>
* <a href="diffhunk://#diff-dcf2294774837d1f40eae9676737cb8c50f6ac940b7c1392660d17d64a84b93aL435-R435">`src/promptflow/promptflow/_cli/_pf/_flow.py`</a>: Added import statements for constants, updated conditions to check flow language, removed empty lines, added commas, and reordered imports. <a href="diffhunk://#diff-dcf2294774837d1f40eae9676737cb8c50f6ac940b7c1392660d17d64a84b93aL435-R435">[1]</a> <a href="diffhunk://#diff-dcf2294774837d1f40eae9676737cb8c50f6ac940b7c1392660d17d64a84b93aR444">[2]</a> <a href="diffhunk://#diff-dcf2294774837d1f40eae9676737cb8c50f6ac940b7c1392660d17d64a84b93aR471">[3]</a> <a href="diffhunk://#diff-dcf2294774837d1f40eae9676737cb8c50f6ac940b7c1392660d17d64a84b93aL40-R40">[4]</a> <a href="diffhunk://#diff-dcf2294774837d1f40eae9676737cb8c50f6ac940b7c1392660d17d64a84b93aL461-R462">[5]</a> <a href="diffhunk://#diff-dcf2294774837d1f40eae9676737cb8c50f6ac940b7c1392660d17d64a84b93aR9-L12">[6]</a>
* <a href="diffhunk://#diff-8edc3f1b002af9c7a0059d7d6ae7a25d1983836b6736d0d7e8c8537c6c3781c2R30">`src/promptflow/promptflow/azure/operations/_run_operations.py`</a>: Modified the import statement to include the `FlowLanguage` constant and updated the condition for checking the language value in the `_resolve_flow` method. <a href="diffhunk://#diff-8edc3f1b002af9c7a0059d7d6ae7a25d1983836b6736d0d7e8c8537c6c3781c2R30">[1]</a> <a href="diffhunk://#diff-8edc3f1b002af9c7a0059d7d6ae7a25d1983836b6736d0d7e8c8537c6c3781c2L783-R784">[2]</a>
* <a href="diffhunk://#diff-afdd40a5d0519512dcf9be48bd46c4caaa2291b808687de77896989af63f47e4R17">`src/promptflow/promptflow/_sdk/operations/_flow_operations.py`</a>: Modified the import statement to include the `LANGUAGE_KEY` constant and updated the key used to access the language value in the `flow.dag` dictionary. <a href="diffhunk://#diff-afdd40a5d0519512dcf9be48bd46c4caaa2291b808687de77896989af63f47e4R17">[1]</a> <a href="diffhunk://#diff-afdd40a5d0519512dcf9be48bd46c4caaa2291b808687de77896989af63f47e4L134-R135">[2]</a>

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_constants.py [hunk 1]
```diff
@@ -66,7 +66,7 @@
 VIS_LIB_CDN_LINK_TMPL = (
     "https://sdk-bulk-test-endpoint.azureedge.net/bulk-test-details/view/{version}/bulkTestDetails.min.js?version=1"
 )
-VIS_LIB_VERSION = "0.0.32"
+VIS_LIB_VERSION = "0.0.33"
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_constants.py:69]: MARK

File: src/promptflow/promptflow/_sdk/operations/_run_operations.py [hunk 2]
```diff
@@ -293,7 +298,20 @@ def _visualize(self, runs: List[Run], html_path: Optional[str] = None) -> None:
...
             details.append(copy.deepcopy(detail))
             metadatas.append(asdict(metadata))
-        data_for_visualize = RunVisualization(detail=details, metadata=metadatas)
+            # TODO: add language to run metadata
+            flow_dag = yaml.safe_load(metadata.dag)
+            configs.append(
+                VisualizationConfig(
+                    [AvailableIDE.VS_CODE]
+                    if flow_dag.get(LANGUAGE_KEY, FlowLanguage.Python) == FlowLanguage.Python
+                    else [AvailableIDE.VS]
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/operations/_run_operations.py:307]: ideally the logic should be: 
 request triggered from vs exetension are targeting vs, not all csharp flow target vs

Comment (elliotzh) [reply src/promptflow/promptflow/_sdk/operations/_run_operations.py:307]: hmm, that will require a further discussion with executor & vs extension on how to pass the information.

Let's track it with another item and merge this PR first: [Task 2835185](https://msdata.visualstudio.com/Vienna/_workitems/edit/2835185): tell sdk where the batch run is triggered to optimize pf run visualize

**Review by wangchao1230 — APPROVED** [review]

**Review by RealKai42 — APPROVED** [review]

File: src/promptflow/promptflow/_sdk/_constants.py (hunk 1 above)
Comment (zhengfeiwang) [inline src/promptflow/promptflow/_sdk/_constants.py:69]: if this upgrade introduces other features than dotnet support, maybe need an update to CHANGELOG

Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/_constants.py:69]: considering you are the release dev who need to take care CHANGELOG in release branch, feel free to do this in this PR or release branch.

**Review by zhengfeiwang — APPROVED** [review]

File: src/promptflow/promptflow/_sdk/_constants.py (hunk 1 above)
Comment (RealKai42) [reply src/promptflow/promptflow/_sdk/_constants.py:69]: There is no change to current python flow

Comment (elliotzh) [reply src/promptflow/promptflow/_sdk/_constants.py:69]: it's an exact copy of 32 for now and they promise that no impact will happen for python user

**Pull request merged**